gxexternalproviders/src/main/java/com/genexus/db/driver/ExternalProviderBox.java

```diff
@@ -0,0 +1,458 @@
+package com.genexus.db.driver;
+
+import com.box.sdk.BoxAPIConnection;
+import com.box.sdk.BoxAPIRequest;
+import com.box.sdk.BoxAPIResponse;
+import com.box.sdk.BoxAPIResponseException;
+import com.box.sdk.BoxConfig;
+import com.box.sdk.BoxDeveloperEditionAPIConnection;
+import com.box.sdk.BoxFile;
+import com.box.sdk.BoxFolder;
+import com.box.sdk.BoxItem;
+import com.box.sdk.BoxResource;
+import com.box.sdk.BoxSearch;
+import com.box.sdk.BoxSearchParameters;
+import com.box.sdk.BoxSharedLink;
+import com.box.sdk.BoxSharedLink.Access;
+import com.box.sdk.BoxSharedLink.Permissions;
+import com.genexus.Application;
+import com.genexus.StructSdtMessages_Message;
+import com.genexus.util.GXService;
+import com.genexus.util.GXServices;
+import com.genexus.util.StorageUtils;
+
+import org.apache.logging.log4j.LogManager;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.net.URL;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ExternalProviderBox implements ExternalProvider {
+	private static final String CONFIG_FILE = "CONFIG_FILE";
+
+	private BoxAPIConnection api;
+
+	private static org.apache.logging.log4j.Logger logger = LogManager.getLogger(ExternalProviderBox.class);
+
+	public ExternalProviderBox() {
+		GXServices services = Application.getGXServices();
+		GXService providerService = services.get(GXServices.STORAGE_SERVICE);
+
+		String configFile = providerService.getProperties().get(CONFIG_FILE);
+		File file = new File(services.configBaseDirectory(), configFile);
+		debugTrace("configFile = \"%s\"", file.getAbsolutePath());
+
+		BoxConfig config;
+		try {
+			Reader reader = new FileReader(file);
+			config = BoxConfig.readFrom(reader);
+		} catch (IOException ex) {
+			throw handleException("read config", ex);
+		}
+
+		api = BoxDeveloperEditionAPIConnection.getAppEnterpriseConnection(config);
+	}
+
+	@Override
+	public void download(String externalFileName, String localFile, boolean isPrivate) {
+		debugTrace("download(\"%s\", \"%s\", %s)", externalFileName, localFile, isPrivate);
+
+		try {
+			BoxFile.Info file = getBoxFileFromPath(externalFileName, isPrivate);
+			OutputStream out = new FileOutputStream(localFile);
+			file.getResource().download(out);
+			out.close();

```

Comment:
This should be closed on finally clause or by using "try with resource".

Reply:
Addressed at next commit. Didn't know the "try with resource" statement.